Feature alexander

Webkonditer opened the pull request.

Webkonditer committed:
Добавил сущности для переписки и отчетов попечителей. Сделал Пет контроллер. Подключил Сваггер.

Webkonditer committed:
Добавлены комментарии в Пет сервисе к новым методам.

Webkonditer committed:
Добавлены репозитории к новым сущностям.

Webkonditer committed:
Добавлены репозитории к новым сущностям.

Webkonditer committed:
Сделаны два метода в волонтерском контроллере.

AIvachkin commented on src/main/java/pro/sky/JD2AnimalShelterBot/controller/PetController.java:
```diff
@@ -0,0 +1,102 @@
+     */
+    @PostMapping
+    public ResponseEntity createPet(@RequestBody Pet pet) {
+        Pet createdPet = petService.createPet(pet);
```
Мне кажется, здесь может выскочить исключение, если pet передать с ненулевым id. Я бы сделал тут проверку.

Webkonditer replied:
В методе Post id игнорируется, благодаря автоинкременту.

AIvachkin commented on src/main/java/pro/sky/JD2AnimalShelterBot/controller/PetController.java:
```diff
@@ -0,0 +1,102 @@
+     */
+    @PutMapping()
+    public ResponseEntity updatePet(@RequestBody Pet pet) {
+        Pet updatePet = petService.updatePet(pet);
```
Здесь, наверное, нужна проверка на то, что pet не null

Webkonditer replied:
согласен

AIvachkin commented on src/main/java/pro/sky/JD2AnimalShelterBot/controller/PetController.java:
```diff
@@ -0,0 +1,102 @@
+     */
+    @DeleteMapping("/{petId}")
+    public ResponseEntity deletePet(@PathVariable Long petId) {
+        petService.delete(petId);
```
И в этом месте проверка, что pet с таким id найден

Webkonditer replied:
Согласен

Webkonditer committed:
Сделаны два эксершена в пет контроллере.

Webkonditer committed:
Сделаны два метода в волонтерском контроллере:
Метод для отправки сообщения пользователю
Метод для получения всей переписки с пользователем
